phenopacketlab-frontend/src/app/component/phenopacket/phenopacket.component.ts

```diff
@@ -30,55 +33,38 @@ import { Interpretation } from 'src/app/models/interpretation';
 export class PhenopacketComponent implements OnInit, OnDestroy {
 
   @Input()
-  phenopacket: Phenopacket;
+  phenopacketId: string;
 
-  @Output() onIdChanged = new EventEmitter<any>();
-  @Output() onSexChanged = new EventEmitter<any>();
-  @Output() onDobChanged = new EventEmitter<any>();
-  @Output() onIndividualChange = new EventEmitter<Individual>();
-
-  summary = '';
-  sex = 'UNKNOWN_SEX';
-  karyotypicSex = 'UNKNOWN_KARYOTYPE';
-  gender: any;
-  // dob: Date;
-  individual: Individual;
-  lastEncounterDate = '';
-
-  status = '';
-  timeOfDeath = '';
-  causeOfDeath: OntologyClass;
-  survivalTime: number;
-
-  active = 'top';
-  viewMode;
-  // accordion
-  step: number;
+  phenopacket: Phenopacket;
 
   ref: DynamicDialogRef;
+  cohortSubscription: Subscription;
 
-  constructor(public dialogService: DialogService, public messageService: MessageService) { }
+  constructor(private cohortService: CohortService,
+    private dialogService: DialogService,
+    private messageService: MessageService) { }
 
   ngOnInit(): void {
-    this.viewMode = 'tab1';
-    if (this.phenopacket) {
-      this.individual = this.phenopacket.subject;
-      this.lastEncounterDate = this.individual.timeAtLastEncounter ? this.individual.timeAtLastEncounter.toString() : '';
-      this.sex = this.individual.sex;
-      this.karyotypicSex = this.individual.karyotypicSex;
-      this.gender = this.individual.gender;
-      // status
-      this.status = this.individual?.vitalStatus?.status?.toString();
-      this.causeOfDeath = this.individual?.vitalStatus?.causeOfDeath;
-      this.timeOfDeath = this.individual?.vitalStatus?.timeOfDeath?.toString();
-      this.survivalTime = this.individual?.vitalStatus?.survivalTimeInDays;
-    }
+    // retrieve phenopacket to edit
+    this.cohortSubscription = this.cohortService.getCohort().subscribe(cohort => {
```

Comment:
Overly complex here. Create a method in the cohort service that will return you a phenopacket by id. Additionally please leverage the rx map/filter features over a for-loop

Reply:
@iimpulse I just committed it: using a pip(map... and find on the phenopacketlist based on id